DatadogCore/Sources/Core/DatadogCore.swift

```diff
@@ -71,6 +71,93 @@ internal final class DatadogCore {
     /// Maximum number of batches per upload.
     internal let maxBatchesPerUpload: Int
 
+    convenience init(
+        configuration: Datadog.Configuration,
+        trackingConsent: TrackingConsent,
+        instanceName: String
+    ) throws {
```

Comment:
💬 This initializer is extracted from what was originally in `Datadog.swift` (behind the static facade). No other code changes were done.

Reply:
May I ask why?

The problem now is that `DatadogCore` depends on `Datadog.*` data types (if we take the clean architecture, this breaks the dependency rule). I would rather have a `DatadogCore` extension defined at the upper level of the module than make `DatadogCore` depend on `Datadog`. If that make sense?

Reply:
✅ done